src/components/registerForm.js

```diff
@@ -0,0 +1,47 @@
+import { useRef, useState } from "react";
+import { registerService } from "../services/userService";
+import {
+  EmailField,
+  NameField,
+  PasswordField,
+  SubmitButton,
+  ErrorField,
+} from "./elements";
+
+function RegisterForm(props) {
+  const [nameError, setNameError] = useState();
+  const [emailError, setEmailError] = useState();
+  const [passwordError, setPasswordError] = useState();
+  const formRef = useRef();
+  const register = () => {
+    let form = new FormData(formRef.current);
+    registerService(form)
+      .then((data) => {
+        setNameError("");
+        setEmailError("");
+        setPasswordError([]);
+        props.changePage("Account created. Please log in");
+      })
+      .catch((errors) => {
+        setNameError("");
+        setEmailError("");
+        setPasswordError([]);
+        if ("name" in errors) setNameError(errors["name"]);
+        if ("email" in errors) setEmailError(errors["email"]);
+        if ("password" in errors) setPasswordError(errors["password"]);
```

Comment:
Setting state is an expensive operation it rerenders screen in newer versions it has been optimized still it's better to use it only when required.
```suggestion
        setNameError("name" in errors ? errors["name"] : "");
        setEmailError("email" in errors ? errors["email"] : "");
        setPasswordError("password" in errors ? errors["password"] : []);
```